rc/lsp.kak

```diff
@@ -177,6 +177,11 @@ define-command -hidden lsp-menu -params 1.. -docstring "Like menu but with promp
             shellquote() {
                 printf "'%s'" "$(printf %s "$1" | sed "s/'/'\\\\''/g; s/§/§§/g; $2")"
             }
+            on_abort=
+            if [ $1 = "-on-abort" ]; then
+                ob_abort="-on-abort $(shellquote "$2" s/¶/¶¶/g)"
```

Comment:
typo (ob_abort -> on_abort)

To make this do something we'd need `prompt -on-abort` support in Kakoune.
Is it actually necessary here though? What's the worst that can happen if we never send our reply?

Reply:
Aside from `metals` waiting forever for a reply, I don't think anything else would happen. 
Btw `prompt -on-abort` has been in Kakoune since https://github.com/mawww/kakoune/commit/2edb7d62ef50c65a04de499768435c2fc41a832c (6 years ago)

Reply:
good, then let's add it to the `prompt` command below, something like

```diff
diff --git a/rc/lsp.kak b/rc/lsp.kak
index 10db851..917a7e7 100644
--- a/rc/lsp.kak
+++ b/rc/lsp.kak
@@ -208,6 +208,7 @@ define-command -hidden lsp-menu -params 1.. -docstring "Like menu but with promp
                     esac
                 ¶
             §" "$cases"
+            printf ' %s' "$on_abort"
             printf ' -menu %s -shell-script-candidates %%§
                 printf %%s %s
                 §\n' "$($kak_opt_lsp_have_kakoune_feature_completion_priority && printf %s -priority)" "$(shellquote "$completion")"
```

which means we don't need shellquote, only kakoune quoting so something like `s/¶/¶¶/g`

Reply:
I added it, but 
> which means we don't need shellquote, only kakoune quoting so something like s/¶/¶¶/g

I'm not sure what you mean here. Are you referring to this part?
```bash
                on_abort="-on-abort $(shellquote "$2" s/¶/¶¶/g)"
```

Reply:
yeah we don't need shellquote for on_abort because on_abort will never be evaluated by a shell (unlike the case statements)